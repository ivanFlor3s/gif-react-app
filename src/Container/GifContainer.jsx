import React, { useState, useEffect } from "react";
import { GifItem } from "../components/GifItem";
import { getGifs } from "../helpers/getGifs";

export const GifContainer = ({ categoria }) => {
  const [gifs, setGifs] = useState([]);

  useEffect(() => {
    getGifs(categoria)
        .then( setGifs )
  }, [categoria]);

  
  return (
    <>
      <h3>{categoria}</h3>
      <div className="card__container">
        {gifs.map((g) => {
          return <GifItem img={g.url} title={g.title} key={g.id}></GifItem>;
        })}
      </div>
    </>
  );
};
