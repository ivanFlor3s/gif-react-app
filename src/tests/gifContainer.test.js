import React from "react";
import { shallow } from "enzyme";
import { GifContainer } from "../Container/GifContainer";
import { useFetchGifs } from "../hooks/useFetchGifs";
jest.mock("../hooks/useFetchGifs");

describe("Test en gif container component", () => {
  test("Pruebo componente Ok", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(
      <GifContainer categoria="Nueva categoria"></GifContainer>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe demostrar items cuando se cargan imagenes con nuestro UseFetchGifs", () => {
    const gifsResult = [
        {id: 'abc', url:'https://localhost/asd.jpg', title:'qti'},
        {id: 'cba', url:'https://localhost/cba.jpg', title:'qtiCba'}
    ]
    useFetchGifs.mockReturnValue({
      data: gifsResult,
      loading: false,
    });
    const wrapper = shallow(
      <GifContainer categoria="Nueva categoria"></GifContainer>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifItem').length).toBe(gifsResult.length)
  });
});
