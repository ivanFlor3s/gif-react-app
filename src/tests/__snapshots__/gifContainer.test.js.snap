// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en gif container component Debe demostrar items cuando se cargan imagenes con nuestro UseFetchGifs 1`] = `
<Fragment>
  <h3>
    Nueva categoria
  </h3>
  <div
    className="card__container"
  >
    <GifItem
      img="https://localhost/asd.jpg"
      key="abc"
      title="qti"
    />
    <GifItem
      img="https://localhost/cba.jpg"
      key="cba"
      title="qtiCba"
    />
  </div>
</Fragment>
`;

exports[`Test en gif container component Pruebo componente Ok 1`] = `
<Fragment>
  <h3>
    Nueva categoria
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Cargando 
  </p>
  <div
    className="card__container"
  />
</Fragment>
`;
